zio/zngio/batch.go

```diff
@@ -33,6 +35,15 @@ func (b *batch) add(r *zed.Value) { b.vals = append(b.vals, *r) }
 
 func (b *batch) Values() []zed.Value { return b.vals }
 
+//XXX
+func (b *batch) NewValue(typ zed.Type, bytes zcode.Bytes) *zed.Value {
+	return zed.NewValue(typ, bytes)
+}
+
+func (b *batch) CopyValue(val zed.Value) *zed.Value {
+	return zed.NewValue(val.Type, val.Bytes)
```

Comment:
If the CopyValue contract is that it returns a value that does not share val.Bytes, then we need an actual copy here.
```suggestion
	return val.Copy()
```

Reply:
same as above